### PR TITLE
QueryGroup & DataSourceSrv & DataSourcePicker changes simplify usage, error handling and reduce duplication, support for uid

### PR DESCRIPTION
**Pull request by torkelo**
This is a continuation of https://github.com/grafana/grafana/pull/29419 

Some really tricky stuff here, think this simplifies some of the complexity around data source selection & error handling. 

* [x] DataSourceSrv: Big changes to unify getting & filtering a list of data sources
* [x] DataSourceSrv: lots o new unit tests
* [x] This simplifies the QueryGroup (previous QueriesTab) component and moves data source & queries into the QueryGroup options
* [x] Moves logic into the DataSourcePicker to simplify and reduce duplication in its usage and share support for uid & error handling 
* [x] Simplifies some logic in QueryEditorRow 
* [x] Test import dashboard logic
* [x] Make sure data source picker in dashboards show the builtIn data sources
* [x] Explore data source picker should not show built-in data sources  
* [x] Better support for the default data source, now the list shows default as a separate entry and you cannot see if your current data source is the default. 
* [x] Updates query variable to use data source picker 
* [x] Update query variable tests & e2e
 

`DataSourceSrv` getExternal and getAll are replace by getList

### CODE REVIEW

File: packages/grafana-runtime/src/services/dataSourceSrv.ts [hunk 1]
```diff
@@ -16,21 +16,27 @@ export interface DataSourceSrv {
...
+/** @public */
+export interface GetDataSourceListFilters {
+  mixed?: boolean;
+  metrics?: boolean;
+  tracing?: boolean;
+  annotations?: boolean;
+  dashboard?: boolean;
+  variables?: boolean;
+  pluginId?: string;
+}
```
Comment (torkelo) [inline packages/grafana-runtime/src/services/dataSourceSrv.ts:38]: @ryantxu not sure how much you care/use DataSourceSrv, this PR refactors this service quite a bit, unifies all the get data source arrays to a single getList function, and a bunch more related to data source selection. 

Leaving getAll and getExternal in the runtime code so plugins using them will still work at runtime

File: e2e/suite1/specs/variables/new-query-variable.ts [hunk 2]
```diff
@@ -46,10 +46,9 @@ describe('Variables - Add variable', () => {
     e2e.pages.Dashboard.Settings.Variables.Edit.QueryVariable.queryOptionsDataSourceSelect()
       .should('be.visible')
       .within(select => {
-        e2e.components.Select.singleValue().should('have.text', '');
+        e2e.components.Select.singleValue().should('have.text', 'gdev-testdata');
```
Comment (hugohaggmark) [inline e2e/suite1/specs/variables/new-query-variable.ts:49]: I guess the default data source is automatically chosen now which makes sense.

**Review by hugohaggmark — APPROVED** [review]
I took the variables out for a spin and it looks so great. I love how the variable actions became less ❤️  So this approval is for the variables, outstanding work ⭐ 

I did find an unrelated bug that needs to be fixed (already on master) when you chose the ADX datasource the VaraibleEditor crashes.

File: e2e/suite1/specs/variables/new-query-variable.ts (hunk 2 above)
Comment (torkelo) [reply e2e/suite1/specs/variables/new-query-variable.ts:49]: yes, exactly

**Review by dprokop — APPROVED** [review]
Quite hard to review this one code-wise but I took it for a spin in Explore/Panel Edit/ Dashboard import and Variebles and it works as expected, haven't noticed any regressions so far.

Great work!

**Pull request merged**